### PR TITLE
Share Post implementation

### PR DESCRIPTION
**Pull request by Yoncity**
**What does this PR do?**
This PR adds the share to social media feature, that enables users to share posts to FaceBook and Twitter

**Description of the task to be completed.**
~ Create SharePost component
~ Create actions that consume the share endpoints
~ Write tests for the component and actions

**How should this be manually tested?**
~ Fetch and checkout to the branch with `git fetch origin ft-share-post && git checkout ft-share-post`
~ Start the server
~ Authenticate with the Google Auth option
~ Press the share icon at the bottom right of any post.

### CODE REVIEW

**Comment by Yoncity** [comment]
[Users should be able to share a post via email, LinkedIn, Facebook and twitter](https://trello.com/c/M6Y7ZG2H/40-users-should-be-able-to-share-a-post-via-email-linkedin-facebook-and-twitter)

File: src/components/SharePost/SharePost.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          <div className="social_share" id="facebook">
+            <EmailShareButton url={POST_URL}>
+              <img
+                src={facebook_logo}
+                className="social_logo"
+                alt="facebook_logo"
+              />
+              <p className="social_name">SHARE ON EMAIL</p>
+            </EmailShareButton>
+          </div> */}
```
Comment (gracelungu) [inline src/components/SharePost/SharePost.jsx]: @Yoncity Please remove the commented code above

File: src/components/SharePost/SharePost.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        </div>
+      </Modal>
+    );
+  }
+}
+
+// const mapStateToProps = state => ({});
```
Comment (gracelungu) [inline src/components/SharePost/SharePost.jsx]: This one too

**Review by gracelungu — CHANGES_REQUESTED** [review]

File: src/components/SharePost/SharePost.jsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import PropTypes from 'prop-types';
+
+import { FRONTEND_BASE_URL } from '../../constants';
+
+import share from '../../actions/sharePostAction';
+
+import {
+  FacebookShareButton,
+  TwitterShareButton,
+  // LinkedinShareButton,
```
Comment (Karlmusingo) [inline src/components/SharePost/SharePost.jsx]: @Yoncity, would you also add share on LinkedIn and Email

File: src/components/SharePost/SharePost.jsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+// const mapStateToProps = state => ({});
+
+const mapDispatchToProps = dispatch => ({
+  share: data => dispatch(share(data)),
+});
+
+SharePost.propTypes = {
+  show: PropTypes.bool,
+  postSlug: PropTypes.string,
```
Comment (Karlmusingo) [inline src/components/SharePost/SharePost.jsx:70]: Also add default value for props which are not required

File: src/components/Feed/Feed.jsx [hunk 5]
```diff
@@ -9,7 +9,14 @@ import ContentLoader from '../Helpers/ContentLoader';
 import onScrollToBottom from '../../helpers/onScrollToBottom';
 import Like from '../../components/Like/Like';
 
+import SharePost from '../SharePost/SharePost';
+
 export class Feed extends Component {
+  state = {
```
Comment (Karlmusingo) [inline src/components/Feed/Feed.jsx]: When there is a constructor, it is better if the state is placed inside the constructor

**Review by Karlmusingo — APPROVED** [review]

**Review by gracelungu — APPROVED** [review]

**Pull request merged**